### ISSUE TITLE
SSD Validation Loss not implemented, a problem?

### ISSUE THREAD

**Issue by caiobarrosv** [dmlc/gluon-cv#1397]
Hello there,

it's actually a question.

Why the validation loss is not implemented in the official gluon-cv tutorial for object detection using SSD?
Link: `https://gluon-cv.mxnet.io/build/examples_detection/train_ssd_voc.html`

It would be easier to detect overfitting by analyzing not only the training loss but also the validation loss, right?

Am I missing something? Is it possible to detect overfitting by only analyzing the training loss and the validation mAP?

Thanks in advance.

**Comment by zhreshold** [comment]
Calculating loss during validation isn't cheap, you can choose to do so but to me validation mAP is a better metric to keep track of

**Comment by caiobarrosv** [comment]
@zhreshold, thank you for the answer 👍 

mAP is indeed a good metric to keep track but not to analyze overfitting.
In order to really identify overfitting we should compare two equal metrics between triain and validation, right?

**Comment by zhreshold** [comment]
You can save the intermediate outputs just as in training and sum the losses if you want to have the equal metrics between train/valid, the idea is to perform the training loop without `autograd` module and trainer so the network won't update itself

**Comment by caiobarrosv** [comment]
@zhreshold,

I will try it out.

Thank you :)
